src/core/componentSystems/StatsSystem.js

```diff
@@ -0,0 +1,59 @@
+import { mrjsUtils } from 'mrjs';
+import { MRSystem } from 'mrjs/core/MRSystem';
+import { MRStats } from 'mrjs/core/entities/MRStats';
+
+import Stats from 'stats.js';
+
+import { HTMLMesh } from 'three/addons/interactive/HTMLMesh.js';
+import { InteractiveGroup } from 'three/addons/interactive/InteractiveGroup.js';
+
+// TODO: JSDoc
+
+export class StatsSystem extends MRSystem {
+    constructor() {
+        super(false);
+        this.statsEntities = [];
+    }
+
+    update() {
+        for (const entity of this.statsEntities) {
+            if (entity.stats === null && this.app) {

```

Comment:
this section should be in the entity's `connected` function - this is the function that gets called when the entity is 'connected to an app (meaning the app exists and <mr-stats> is in the html within it)

[mr-hyperlink's use of connected ](https://github.com/Volumetrics-io/mrjs/blob/main/src/core/entities/MRHyperlink.js#L39)is a good example of this

Reply:
to clarify, when i say 'this section' here i mean within the `if (entity.stats...) { }`

Reply:
Any proper way to access `app` from `connected()`? Because of dependency with `app` I wrote this initialization in system.

Reply:
not directly, but a decent workaround is: 

you leave the main 'constructor' the implementation that doesnt rely on the dependency directly into the entity's own 'constructor' or 'connected' functions (whichever makes most sense in your opinion)

and then you move the part of this 'constructor' implementation that relies on mr-app to the `onNewEntity` section (kind of like [how we do it in LayoutSystem](https://github.com/Volumetrics-io/mrjs/blob/main/src/core/componentSystems/LayoutSystem.js#L34))

Reply:
And, with the newer Three.js rev, the initialization will have a dependency with `app.user` initialized at https://github.com/Volumetrics-io/mrjs/blob/15cc832a595506488906a4d97fe9c2055ea1e86d/src/core/MRApp.js#L125 Any good practice to initialize a resource with `app.user`?

Reply:
that's a good question - for reference:
- systems are only there if an app exists: [ref](https://github.com/Volumetrics-io/mrjs/blob/main/src/core/MRApp.js#L108)
- [engine-started](https://github.com/Volumetrics-io/mrjs/blob/979ea75652b135e175296fb05b4159b42a9d7a96/src/utils/Physics.js#L24) is triggered by the physics world being created properly by rapier
- which triggers the creation of the [remaining systems](https://github.com/Volumetrics-io/mrjs/blob/main/src/core/MRApp.js#L124)
- each system then runs through the scene for each entity and touches them one by one
that is - since `<mr-app>` exists, we hit the `onNewEntity` function within whatever system is being called
- meaning it is guaranteed the `app` is ready when the `onNewEntity` function for the system is hit

let me know if that makes sense - we still need to add better documentation for how our ECS rundown works in MRjs in docs.mrjs.io, but I tried to use code-callouts for reference

Reply:
[for this question](https://github.com/Volumetrics-io/mrjs/pull/508#discussion_r1529447438) - this just means that it's a better case to have that part of the entity initialized through `onNewEntity` of the associated system

there are three places where an entity can/should be setup:
- its own constructor
- its own `connected` function
- a system's `onNewEntity` function

the more scope that is needed, the more you go down the bulleted list